app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -195,7 +195,9 @@ def update_host_status
         host = Host.find(input[:host][:id])
         status = host.execution_status_object || host.build_execution_status_object
         status.status = exit_status.to_s == "0" ? HostStatus::ExecutionStatus::OK : HostStatus::ExecutionStatus::ERROR
+        status.refresh
```

Comment:
Probably because it's operating on a non-saved object. Isn't it possible to only save a single attribute, instead of the whole object? I'm guessing `build_execution_status_object` instantiates a new object.

Question is, why does it need to refresh in the first place? Line 195 implies the host object is a fresh object so what are the changes it actually has an outdated cache?

Reply:
I would be happy to only add update_timestamp https://github.com/theforeman/foreman/blob/develop/app/models/host_status/status.rb#L78 but its private.

Should it maybe only call "status.refresh" in case of a already existing execution_status_object?